AElf.Consensus/ConsensusService.cs

```diff
@@ -32,57 +36,84 @@ public class ConsensusService : IConsensusService
 
         public ConsensusService(IConsensusObserver consensusObserver, IExecutingService executingService,
             IConsensusInformationGenerationService consensusInformationGenerationService,
-            IAccountService accountService)
+            IAccountService accountService, ITransactionExecutingService transactionExecutingService,
+            IBlockchainService blockchainService)
         {
             _consensusObserver = consensusObserver;
-            _executingService = executingService;
             _consensusInformationGenerationService = consensusInformationGenerationService;
             _accountService = accountService;
+            _transactionExecutingService = transactionExecutingService;
+            _blockchainService = blockchainService;
 
             Logger = NullLogger<ConsensusService>.Instance;
         }
 
-        public async Task TriggerConsensus(int chainId)
-        {
+        public async Task TriggerConsensusAsync(int chainId)
+        {            
+            var chain = await _blockchainService.GetChainAsync(chainId);
+            var chainContext = new ChainContext
+            {
+                ChainId = chainId,
+                BlockHash = chain?.BestChainHash ?? Hash.Genesis,
```

Comment:
do not use "chain?"

it should throw exception

Reply:
> do not use "chain?"
> 
> it should throw exception

If this node is the first node, the chain is null because it is him to initial the chain.